### PR TITLE
SOS2.0 and WaterML2.0

### PR DESCRIPTION
**Pull request by kwilcox**
This is a manual rebase of https://github.com/geopython/OWSLib/pull/171 onto master along with a few small fixes in the PR.

### CODE REVIEW

**Comment by tomkralidis** [comment]
@kwilcox thanks for this.  Looks like some [associated test failures](https://travis-ci.org/geopython/OWSLib/builds/63681696)?

**Comment by kwilcox** [comment]
@tomkralidis all service endpoint test failures now

**Comment by peterataylor** [comment]
Hi guys,

Any updates on this? The build failures were due to non responsive services called from the doc tests. Do you want to try and capture potential fails or change the service endpoint in the doctest? The general issue is service stability + longevity - can't really be sure all test services will be up.

**Comment by kwilcox** [comment]
Slipped off my radar. I can rebase onto `master` so it can be merged.  The test services going down is a long standing issue... I just opened an issue for it: https://github.com/geopython/OWSLib/issues/312

**Comment by peterataylor** [comment]
Thanks Kyle. That would be great if you could rebase; let me know if I can help.

**Comment by kwilcox** [comment]
@peterataylor Do you know of any SOS2.0 servers I can rewrite the tests against? It took me so long... they have all gone away.

**Comment by tomkralidis** [comment]
@kwilcox here's one: http://ngwd-bdnes.cits.nrcan.gc.ca/GinService/sos/gw

**Comment by peterataylor** [comment]
Thanks @kwilcox . Actually I've recently been testing a new SOS2.0 service at the Bureau of Met here: http://www.bom.gov.au/waterdata/service. I have a py notebook using it here: http://nbviewer.jupyter.org/github/peterataylor/bom-water-data-service-example/blob/master/bom-water-sos-example.ipynb. 

This is likely to be quite a stable service - so a good candidate for the test suite. I can help to integrate it if need be.

**Comment by domlowe** [comment]
Good idea Pete. Typo in the URL - should be:
http://www.bom.gov.au/waterdata/services

**Comment by kwilcox** [comment]
@peterataylor I'm not having much luck with that service... does your notebook work? Do you have more commits to add?

**Comment by peterataylor** [comment]
Hi @kwilcox - yes sorry there were two additional changes I had to make to get it fully working: https://github.com/peterataylor/OWSLib/commit/9274f8e1f06512d3ff2a7ed966d67b44f266309e. One is the way the Bureau service describes its endpoints in the caps document - the base URL is incorrect. I had to put in a temporary hack in to get it to work. I've reported the issue. We obviously don't want this fix in the library, so may be a blocker until we can get the service fixed..?

The second was adding observed_properties to the service metadata where the observed properties are not contained within an offering (they are inherited across all offerings).

**Comment by kwilcox** [comment]
Please don't merge, waiting for other PRs to be merged first and ill rebase

**Comment by kwilcox** [comment]
This is ready to merge. The test failures are all broken or changing external services (which we really need to fix).

File: owslib/iso.py [hunk 1]
```diff
@@ -654,6 +654,9 @@ def __init__(self, md=None):
             val = md.find(util.nspath_eval('srv:accessProperties/gmd:MD_StandardOrderProcess/gmd:fees/gco:CharacterString', namespaces))
             self.fees = util.testXMLValue(val)
 
+            val = md.find(util.nspath_eval('srv:accessProperties/gmd:MD_StandardOrderProcess/gmd:fees/gco:CharacterString', namespaces))
```
Comment (tomkralidis) [inline owslib/iso.py]: is this a duplicated by accident?

File: setup.py [hunk 2]
```diff
@@ -14,7 +14,7 @@ def run_tests(self):
         errno = pytest.main(self.test_args)
         sys.exit(errno)
 
-readme = open('README.txt').read()
```
Comment (tomkralidis) [inline setup.py]: Please keep README.txt as is as this will create packaging / manifest problems downstream.

File: owslib/iso.py (hunk 1 above)
Comment (kwilcox) [inline owslib/iso.py]: Yup bad merge, will remove

File: setup.py (hunk 2 above)
Comment (kwilcox) [inline setup.py]: I didn't even notice this!  I'll revert this to the .txt

**Comment by tomkralidis** [comment]
Squashed and merged -- thanks @peterataylor / @kwilcox !

**Pull request merged**